### PR TITLE
TST Fixes random seed for flaky test_keras

### PR DESCRIPTION
**Pull request by thomasjpfan**
Follow up to #794

I suspect the failing test on `main` is due to sampling of the parameter space that may lead to scores that are constant.

CC @stsievert

### CODE REVIEW

File: tests/model_selection/test_keras.py [hunk 1]
```diff
@@ -56,7 +58,12 @@ def test_keras(c, s, a, b):
     params = {"lr": loguniform(1e-3, 1e-1)}
```
Comment (stsievert) [inline tests/model_selection/test_keras.py:58]:
```suggestion
    params = {"lr": loguniform(1e-3, 1e-1), "random_state": [1]}
```

If we're fixing the random state, why not in the model too? This fixes the initialization (not the index ordering of the optimizer).

**Review by stsievert — COMMENTED** [review]
Yeah, `random_state` is producing the errors. I'm not seeing why else the test would be non-deterministic.

**Comment by stsievert** [comment]
Hm.... looks like the error has to do with timeout on windows. I'm not seeing how that's induced by this PR.

**Comment by thomasjpfan** [comment]
The same test is also failing on `main`: https://dev.azure.com/dask-dev/dask/_build/results?buildId=1961&view=logs&jobId=2eab4704-62a3-55d0-a524-46b534a6e811&j=2eab4704-62a3-55d0-a524-46b534a6e811&t=24a2918a-3d47-54f3-09aa-2db1b58035d7

**Review by stsievert — APPROVED** [review]
